ddl/partition.go

```diff
@@ -1446,3 +1448,42 @@ func truncateTableByReassignPartitionIDs(t *meta.Meta, tblInfo *model.TableInfo)
 	tblInfo.Partition.Definitions = newDefs
 	return nil
 }
+
+func onAlterTablePartition(t *meta.Meta, job *model.Job) (int64, error) {
+	var partitionID int64
+	var rules []*placement.Rule
+	err := job.DecodeArgs(&partitionID, &rules)
+	if err != nil {
+		job.State = model.JobStateCancelled
+		return 0, errors.Trace(err)
+	}
+
+	ver, err := t.GetSchemaVersion()
```

Comment:
how about move this after all this check and sync work (like L1486)

Reply:
Not a good idea. I thought again, what if it fails to get schema version at L1486? Retry the job again? But the job is already done.

Reply:
Or you can put it after the local check, and before sync work

Reply:
Sounds reasonable.